prov/verbs/src/verbs_ep.c

```diff
@@ -121,17 +121,34 @@ void vrb_shutdown_qp_in_err(struct vrb_ep *ep)
 	if (!ep || !ep->ibv_qp || ep->ibv_qp->state == IBV_QPS_UNKNOWN || !ep->eq)
```

Comment:
ep->ibv_qp->state == IBV_QPS_UNKNOWN can be replaced with !ep->shutdown and setting ep->ibv_qp->state = IBV_QPS_UNKNOWN in line 144 can be removed.